app/src/main/java/org/simple/clinic/patient/Patient.kt

```diff
@@ -200,5 +201,36 @@ data class Patient(
         @Embedded(prefix = "businessid_")
         val businessId: BusinessId?
     )
+
+    @Query("""
+    SELECT (
+            CASE
+              WHEN
+              (BP.systolic > 140 OR BP.diastolic > 90
+              OR MH.hasHadHeartAttack = :yesAnswer
+              OR MH.hasHadStroke = :yesAnswer
+              OR MH.hasDiabetes = :yesAnswer
+              OR MH.hasHadKidneyDisease = :yesAnswer
+              OR PD.uuid IS NOT NULL)
+              AND A.uuid IS NULL
+              AND (BP.createdAt < :lastRecordedBpThreshold OR BP.createdAt IS NULL)
+              THEN 1
+              ELSE 0
+            END
+          ) AS isPatientDefaulter
+          FROM Patient P
+          LEFT JOIN (SELECT * FROM BloodPressureMeasurement BP WHERE BP.deletedAt IS NULL AND BP.patientUuid = :patientUuid ORDER BY BP.createdAt DESC LIMIT 1) BP
+          LEFT JOIN (SELECT * FROM MedicalHistory MH WHERE MH.deletedAt IS NULL AND MH.patientUuid = :patientUuid ORDER BY MH.updatedAt DESC LIMIT 1) MH
+          LEFT JOIN (SELECT * FROM PrescribedDrug PD WHERE PD.deletedAt IS NULL AND PD.patientUuid = :patientUuid ORDER BY PD.updatedAt DESC LIMIT 1) PD
+          LEFT JOIN Appointment A ON(A.patientUuid = P.uuid AND A.deletedAt IS NULL AND A.status = :scheduled)
+          WHERE P.uuid = :patientUuid AND P.deletedAt IS NULL
+
```

Comment:
Instead of using `*` to fetch all the fields from the tables, can we just fetch the ones that are necessary to compute whether a patient is a defaulter or not?

Reply:
Done.